sdk/core/core/inc/az_span_builder.h

```diff
@@ -63,6 +63,13 @@ AZ_NODISCARD az_result az_span_builder_append_byte(az_span_builder * const self,
 
 AZ_ACTION_FUNC(az_span_builder_append, az_span_builder, az_span_action)
 
+/**
+ * Replace all contents from a starting position to an end position with the content of a provided
+ * span
+ */
+AZ_NODISCARD az_result
+az_span_builder_replace(az_span_builder * const self, size_t start, size_t end, az_span const span);
```

Comment:
Instead of taking a start and end, should this take a current_span with a new_span?

Reply:
you mean some kind of `find_and_replace`?  or  `replace_all?`
For `replace_all` we would use az_span_builder_replace(builder, 0, builder.size, new_span);
for `find_and_replace` we would first need to introduce new function `index_of`
```c
size_t index_of;
result = az_span_find(source_span, find_this_span, index_of);
// then replace
az_span_builder_replace(builder, index_of, index_of + find_this_span.size, new_span)
```

This `replace function` (maybe we need other name) will calculate any required shifting right or left. Main idea here is to use this function as general operation for:
-  insert span at any position by shifting right
-  override span content:
-  remove / clear
-  replace
-  something else

Reply:
I mean the signature of replace could be like this:
az_span_builder_replace(
   az_span_builder * const self,
   az_span current_span,
   az_span const new_span);

A span IS A start and size and so why pass start & size as separate parameters.

Here's another thought about this: let's say we have a find function that takes a big span as its 1st parameter and short span as it's second parameter. The function searches in the big span for the short span and this function returns a span indicating where the short span is within the big span. Now, we want to replace this span with some new span by calling your replace function. I want to pass to replace the span returned from find as the current_span parameter.